WordPressAuthenticator/GoogleSignIn/GoogleClientId.swift

```diff
@@ -0,0 +1,26 @@
+public struct GoogleClientId {
+
+    let value: String
+
+    public init?(string: String) {
+        guard string.split(separator: ".").count > 1 else {
+            return nil
+        }
+        self.value = string
+    }
+
+    /// See https://developers.google.com/identity/protocols/oauth2/native-app#step1-code-verifier
+    func redirectURI(path: String?) -> String {
+        let root = value.split(separator: ".").reversed().joined(separator: ".")
+
+        guard let path else {
+            return root
+        }
+
+        return "\(root):/\(path)"
+    }
+
+    var defaultRedirectURI: String {
+        redirectURI(path: "oauth2callback")
+    }
+}
```

Comment:
I decided to create a dedicated type wrapping `String` for the Google Client Id value so that we could have validation and utility methods encapsulated in it.

What do you think of this approach? I might use it for other parts of this new functionality.

Reply:
By the way, @crazytonyli, I haven't forgotten your [suggestion](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/710#discussion_r1035358491) about using a different value for the redirect URI. But I'm keeping that as a later-on task to prioritize having something working end-to-end.